### PR TITLE
SIGN_MODE_DIRECT

### LINKED ISSUE #212: Add support for SIGN_MODE_DIRECT
After #211 we need to create a `SignDoc` and `TxRaw` in order to submit a binary tx to Tendermint.

This can either be tested by using curl to send the tranyaction to Tendermint (https://docs.tendermint.com/master/rpc/) or by doing #204 first

### PR DESCRIPTION
**Pull request by willclarktech**
Opening here to avoid conflicts on 212-sign-mode-direct.

Closes #212 

Signatures are different despite the transaction objects otherwise being identical. I'll investigate this afternoon.

### CODE REVIEW

File: packages/proto-signing/package.json [hunk 1]
```diff
@@ -44,6 +44,8 @@
     "postdefine-proto": "prettier --write \"src/generated/codecimpl.*\""
   },
   "dependencies": {
+    "@cosmjs/launchpad": "^0.22.0",
+    "@cosmjs/tendermint-rpc": "^0.22.0",
```
Comment (webmaster128) [inline packages/proto-signing/package.json:48]: - Do we really need launchpad?
- Shouldn't those be dev dependencies?

File: packages/proto-signing/src/signing.spec.ts [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+              },
+            ],
+          },
+        },
+      ],
+    };
+    const txBodyBytes = myRegistry.encode({
+      typeUrl: "/cosmos.tx.TxBody",
+      value: txBodyFields,
+    });
```
Comment (webmaster128) [inline packages/proto-signing/src/signing.spec.ts:132]: Does this do the two step serialization of messages? A message is added to TxBody as raw bytes

File: packages/proto-signing/package.json (hunk 1 above)
Comment (willclarktech) [reply packages/proto-signing/package.json]: We use launchpad for the wallet. It seems pretty convenient.

File: packages/proto-signing/src/signing.spec.ts (hunk 2 above)
Comment (willclarktech) [reply packages/proto-signing/src/signing.spec.ts:132]: Yes: https://github.com/CosmWasm/cosmjs/blob/master/packages/proto-signing/src/registry.ts#L65-L90

File: packages/proto-signing/src/msgs.ts [hunk 3]
```diff
@@ -26,3 +26,19 @@ export class MsgSend extends Message {
   @cosmosField.repeatedMessage(3, Coin)
   public readonly amount?: readonly Coin[];
 }
+
+// TODO: move somewhere else as needed
+@registered(defaultRegistry, "/cosmos.auth.BaseAccount")
+export class BaseAccount extends Message {
```
Comment (webmaster128) [inline packages/proto-signing/src/msgs.ts:32]: This type is now in the autogenerated code in @cosmjs/starport (where all the queries are) and not needed here anymore

**Pull request merged**